map.js

```diff
@@ -0,0 +1,68 @@
+import L from "leaflet";
+import {
+  loadWeatherForecast,
+  failedForecast,
+  hideForecastLoader,
+} from "./domManipulation";
+import fetchWeatherForecast from "./api";
+
+const map = L.map("map").setView([-28.921631, 25.224609], 4);
+// Set the initial marker location ot that of Pretoria as this is the default.
+const marker = L.marker([-25.73134, 28.21837]).addTo(map);
+
+export function loadMap() {
+  L.tileLayer("https://tile.openstreetmap.org/{z}/{x}/{y}.png", {
+    maxZoom: 19,
+    attribution:
+      "&copy; <a href=\"http://www.openstreetmap.org/copyright\">OpenStreetMap</a>",
+  }).addTo(map);
+  return map;
+}
+export function loadMarker(lat, long) {
+  marker.setLatLng([lat, long]).addTo(map);
+  map.setView([lat, long], 8);
+}
+
+// Controls the pop-up for when a user selects a location on the map;
+export function togglePopup(customLat, customLong) {
+  document.getElementById("popup-loading-container").style.visibility = "visible";
```

Comment:
The map shouldn't know about the rest of the DOM

Reply:
Moved to dommanipulation.js.